libs/api_common/windows_platform_common.cpp

```diff
@@ -123,6 +123,88 @@ set_program_under_verification(ebpf_handle_t program)
     _program_under_verification = program;
 }
 
+static ebpf_result_t
+_duplicate_program_info(_In_ const ebpf_program_info_t* info, _Outptr_ ebpf_program_info_t** new_info)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_program_info_t* program_info = nullptr;
+    ebpf_helper_function_prototype_t* global_helpers = nullptr;
+    ebpf_helper_function_prototype_t* program_type_specific_helpers = nullptr;
+    ebpf_program_type_descriptor_t* program_type_descriptor = nullptr;
+    ebpf_context_descriptor_t* context_descriptor = nullptr;
+
+    program_info = (ebpf_program_info_t*)ebpf_allocate(sizeof(ebpf_program_info_t));
+    if (program_info == nullptr) {
```

Comment:
This allocation is leaked if the next allocation fails?

Reply:
There is a call to `ebpf_program_info_free(program_info);` at 203 which should free this memory in case of any subsequent failure.